test/gst-vaapi/transcode/transcoder.py

```diff
@@ -11,45 +11,199 @@
+      plats, reqs, ffparams = self.requirements[ttype].get(
+        codec, {}).get(mode, ([], (False, False), None))
+      return plats, reqs, ffparams
 
-  def transcode_1to1(self):
-    self.transcoded = get_media()._test_artifact(
-      "{case}_{width}x{height}_{mode}.{dstextension}".format(**vars(self)))
-    get_media().test_call_timeout = vars(self).get("call_timeout", 0)
-    call("gst-launch-1.0 -vf filesrc location={source} ! {gsttrans}"
-      " ! filesink location={transcoded}".format(**vars(self)))
+  def validate_spec(self):
+    from slash.utils.pattern_matching import Matcher
 
-    self.convert_toyuv()
-    self.check_metrics()
+    assert len(self.outputs), "Invalid test case specification, outputs data empty"
+    assert self.mode in ["sw", "hw"], "Invalid test case specification as mode type not valid"
 
-  def convert_toyuv(self):
-    self.decodedYUV = get_media()._test_artifact(
-       "{case}_{width}x{height}_{mode}_{trans_type}.yuv".format(**vars(self)))
-    call("gst-launch-1.0 -vf filesrc location={transcoded}"
-      " ! {gstdecoder2}"
-      " ! videoconvert ! video/x-raw,format=I420"
-      " ! checksumsink2 file-checksum=false qos=false"
-      " frame-checksum=false plane-checksum=false dump-output=true"
-      " dump-location={decodedYUV}".format(**vars(self)))
+    # generate platform list based on test runtime parameters
+    iplats, ireq, _ =  self.get_requirements_data("decode", self.codec, self.mode)
+    platforms = set(iplats)
+    requires = [ireq,]
+
+    for output in self.outputs:
+      codec = output["codec"]
+      mode  = output["mode"]
+      assert mode in ["sw", "hw"], "Invalid test case specification as output mode type not valid"
+      oplats, oreq, _ = self.get_requirements_data("encode", codec, mode)
+      platforms &= set(oplats)
+      requires.append(oreq)
+
+    # create matchers based on command-line filters
+    matchers = [Matcher(s) for s in slash.config.root.run.filter_strings]
+
+    # check if user supplied any platform tag on command line
+    pmatch = any(map(lambda p: any([m.matches(p) for m in matchers]), ALL_PLATFORMS))
+
+    # if user supplied a platform tag, check if this test can support it via the
+    # test param-based required platforms
+    if pmatch and not any(map(lambda p: any([m.matches(p) for m in matchers]), platforms)):
+      slash.skip_test("unsupported platform")
+
+    # check required
+    if not all([t for t,m in requires]):
+      slash.skip_test("One or more software requirements not met")
+
+  def gen_input_opts(self):
+    opts = " gst-launch-1.0 -vf filesrc location={source}"
+    return opts.format(**vars(self))
+
+  def gen_output_opts(self):
+    self.goutputs = dict()
+
+    opts = ""
+    _, _, ffdecparms = self.get_requirements_data("decode", self.codec, self.mode)
+    assert ffdecparms is not None, "gen_output_opts decode parameters are empty"
+    ffdecoder, ffdectype = ffdecparms
```

Comment:
decode params should be setup in `gen_input_opts`

Reply:
moved to gen_input_opts